backend/gtime/gtime.go

```diff
@@ -75,18 +75,45 @@ func ParseDuration(inp string) (time.Duration, error) {
 }
 
 func parse(inp string) (time.Duration, string, error) {
-	result := dateUnitPattern.FindSubmatch([]byte(inp))
+	// Fast path for simple duration formats (no date units)
+	if len(inp) > 0 {
```

Comment:
I think this should be:
```
  if inp == "" {
    return 0, "", errors.New("empty input")
  }
```
then the following code can come out of the if block.